javascript/index.js

```diff
@@ -1,20 +1,33 @@
-const WebSocket = require('ws');
-const socket = new WebSocket("wss://stream.aisstream.io/v0/stream")
+import WebSocket from "ws";
+const socket = new WebSocket("wss://stream.aisstream.io/v0/stream");
+const API_KEY = process.env.AISSTREAM_API_KEY; # Would need to be established first
+socket.addEventListener("open", (_) => {
+  const subscriptionMessage = {
+    APIkey: API_KEY,
+    BoundingBoxes: [
+      [
+        [-180, -90],
+        [180, 90],
+      ],
+    ],
+  };
+  console.log(JSON.stringify(subscriptionMessage));
+  socket.send(JSON.stringify(subscriptionMessage));
+});
 
-socket.onopen = function (_) {
-    let subscriptionMessage = {
-        Apikey: "<YOUR API KEY>",
-        BoundingBoxes: [[[-180, -90], [180, 90]]]
-    }
-    socket.send(JSON.stringify(subscriptionMessage));
-};
+socket.addEventListener("error", (event) => {
+  console.log(event);
+});
 
-socket.onmessage = function (event) {
-    let aisMessage = JSON.parse(event.data)
-    if (aisMessage["MessageType"] === "PositionReport") {
-        let positionReport = aisMessage["Message"]["PositionReport"]
-        console.log(`ShipId: ${positionReport["UserID"]} Latitude: ${positionReport['Latitude']} Latitude: ${positionReport['Longitude']}`)
-    }
+socket.addEventListener("message", (event) => {
```

Comment:
Could you explain what the advantage is to use event listeners verse `onmessage` ? We are not js developers by any means so are not familiar with the pros/cons of this approach.

Reply:
The main difference is, that you can add multiple handlers for an event with `addEventListener`. The on___ property or attribute allows to register only one handler. Setting on___ will remove all existing handlers.